docs/decisions/0010-limit-modulestore-use-in-lms.rst

```diff
@@ -0,0 +1,273 @@
+Unfortunately, the Modulestore is a large, complex system that is shared by many
+features. This has been the cause of many bugs and performance issues over the
+years. Modulestore performance has been gradually improved over the years, but
+at the cost of adding even more complexity.
+
+Decisions
+=========
+
+1. New features will be implemented without making calls to the Modulestore in
+   the LMS.
+2. Teams should opportunistically remove LMS Modulestore dependencies in
+   existing features as they are being worked on.
+3. Apps may still access the Modulestore in Studio.
+4. Apps may make use of newer, more limited LMS APIs that can answer some of the
+   same questions. This includes CourseOverviews for settings metadata and
+   Learning Sequences for course outlines.
+5. Apps that need other course content data should push that data into their own
+   data models during course publish, by listening to the course_published
+   signal and launching a celery task that will query the Modulestore in the
+   Studio process.
+
+Goals
+=====
+
+**Application correctness is easier to reason about.**
+  Modulestore often has obscure edge cases like non-standard course hierarchies,
+  inheritable attributes set in places where Studio would not normally offer it
+  as an option, A/B tests at the section level, etc.
+
+**Application behavior is more predictable.**
+  Modulestore grabs large chunks of the course at once, leading to major
+  variability in performance in small vs. large courses, or in courses where
+  certain advanced features have been enabled. Serving LMS requests from simpler
+  application data models will make the operational behavior much more
+  predictable.
+
+**Tests are easier to write and run faster.**
+  CourseOverviews and UserCourseOutlineData objects are much easier to create
+  and mock for testing purposes than a tree of XBlocks, and do not suffer from
+  complicated publishing rules. Using the modulestore also imposes a substantial
+  performance penalty when creating and modifying courses, making modulestore
+  access the dominant part of the edx-platform test suite running time.
+
+**Applications are more resilient to user-facing failures.**
+  Many features today are at least partially implemented in Modulestore and the
+  XBlocks that it returns. Changes in those features can cause bugs that ripple
+  out into completely unrelated features being developed by other teams.
+
+  If an LMS application queries the Modulestore when responding to a user
+  request, then unexpected failures in shared Modulestore code will cause
+  user-facing errors. A feature that reads Modulestore data at the time of
+  course publish and pushes it into its own data model will still be affected by
+  Modulestore-related bugs, but it will result in stale data and not an entirely
+  broken experience.
+
+**We can reduce the size of the edx-platform monolith.**
```

Comment:
What do you think about mentioning LMS/CMS decoupling, either as a note within this goal or its own goal?